andela/travel_tool_front
# #161084713 Fix Checkbox on Personal Details section

#### What does this PR do?

Fixes the checkbox bug in the personal details section which changes user profile when the checkbox is checked but the request is not sent.

#### Description of Task to be completed?

- Pre-fill User Profile form with current values from the database (redux store)
- Pre-fill Personal Details form in Request modal with current values from the database (redux store)
- Don't make updates to the user profile when the checkbox is checked, provided the user did not submit any request
- Don't change user profile when the user makes changes to personal details and checkbox is unchecked
- Update user profile when the user makes changes to personal details and the checkbox is checked
- Update to the user profile is by-passed if the user hasn't made any changes to the personal details
 and the checkbox is checked when the  ***Send  Request*** button is clicked. 
- Refactor tests

#### How should this be manually tested?

##### Prerequisites 
You should have your back-end server running

- Using your terminal, clone this repo -``` git clone https://github.com/andela/travel_tool_front.git``` 
- Change directory to the repo in your text editor-```cd travela_tool_front```
- Open the project in your text editor
- Checkout to this branch-```git checkout bg-fix-request-modal-checkbox-161084713```
- Install the dependencies-```yarn install```
- Start the server-```yarn start ```
- Navigate to localhost:3000 in your browser
- Login to the application

##### Testing when checkbox is clicked

- Navigate to the requests page
- Make a new request by clicking on the new request button
- If you have set up your user profile, the personal details section should be prefilled with your information. If not, the fields should be empty.
- Make changes to the personal details fields
- Check the checkbox
- Close the request form
- Re-open the request form
- There should be no changes made to the personal details section

- Uncheck the checkbox to make changes to the personal details fields 
- Check the checkbox  once again
- Fill out the rest of the form and submit a request
- When request is creates successfully, click on the new request button
- The personal details section should reflect the changes you made

- Navigate to settings 
- Navigate to the user profile page
- The user profile page should also reflect the changes made to the personal details

##### Testing when checkbox is not clicked

- Navigate to the requests page
- Make a new request by clicking on the new request button
- Uncheck the checkbox
- Make changes to the personal details fields
- Fill out the rest of the form and submit a request
- When request is creates successfully, click on the new request button
- The personal details section should not reflect the changes you made

- Navigate to settings 
- Navigate to the user profile page
- The user profile page should not reflect the changes made to the personal details

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161084713](https://www.pivotaltracker.com/n/projects/2184887/stories/161084713)

#### Screenshots (if appropriate)

<img width="1280" alt="screen shot 2018-10-21 at 12 53 27 pm" src="https://user-images.githubusercontent.com/13402544/47265451-b8c70f80-d530-11e8-9198-2fa00d4c94c3.png">
<img width="1280" alt="screen shot 2018-10-21 at 12 53 41 pm" src="https://user-images.githubusercontent.com/13402544/47265454-bb296980-d530-11e8-903f-10cb34b3e9ab.png">
